pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2260,20 +2294,32 @@ func (s *AWSCloud) GetLoadBalancer(service *api.Service) (*api.LoadBalancerStatu
 		return nil, false, nil
 	}
 
-	status := toStatus(lb)
+	status := s.toStatus(lb, service)
 	return status, true, nil
 }
 
-func toStatus(lb *elb.LoadBalancerDescription) *api.LoadBalancerStatus {
-	status := &api.LoadBalancerStatus{}
+func (s *AWSCloud) toStatus(lb *elb.LoadBalancerDescription, apiService *api.Service) *api.LoadBalancerStatus {
+	var ingress []api.LoadBalancerIngress
 
 	if !isNilOrEmpty(lb.DNSName) {
-		var ingress api.LoadBalancerIngress
-		ingress.Hostname = orEmpty(lb.DNSName)
-		status.Ingress = []api.LoadBalancerIngress{ingress}
+		var dnsIngress api.LoadBalancerIngress
+		dnsIngress.Hostname = orEmpty(lb.DNSName)
```

Comment:
`aws.StringValue` does the same as `orEmpty`.  I try to fix when I'm touching lines that orEmpty.  Up to you whether you think it's worth it here.